invenio_rdm_records/services/components/pids.py

```diff
@@ -212,6 +212,11 @@ def delete_record(self, identity, data=None, record=None, uow=None):
             self.service.pids.parent_pid_manager.discard_all(
                 parent_pids, soft_delete=True
             )
+        else:
+            # We're sending a task in case there is a race condition with two
+            # versions being deleted at the same time to ensure that we have
+            # consistent database state
+            self.uow.register(TaskOp(cleanup_parent_pids, record["id"]))
```

Comment:
Based on the `signature`, the `record` param could be `None`?

And more important, we are passing the `record id` to the task, but it looks like that the task expects the record obj instead.